1507-Revise-Theme-Selector-Controls

giduac opened the pull request:
1507-Revise-Theme-Selector-Controls
[issue 1507](https://github.com/Krypton-Suite/Standard-Toolkit/issues/1507)
####  First part:
KThemeCBB and common code for the other selectors.
This is the blueprint for the other switchers.
The interface definition for now is in KThemeCBB.cs. Will move that later on to another file and location.

The source files of KThemeCBB & Common code are included as a zip. So you can have a better look.
[source.zip](https://github.com/user-attachments/files/15569463/source.zip)

Small Demo
https://github.com/Krypton-Suite/Standard-Toolkit/assets/96108132/f3794534-1603-4be3-a923-31b867e2e47d

![compile-results](https://github.com/Krypton-Suite/Standard-Toolkit/assets/96108132/99446dd4-951f-4ae4-98ec-0b5c64690fc4)

Smurf-IV commented on Source/Krypton Components/Krypton.Toolkit/General/CommonHelperThemeSelectors.cs:
```diff
@@ -0,0 +1,120 @@
+﻿#region BSD License
+/*
+ *
+ * Original BSD 3-Clause License (https://github.com/ComponentFactory/Krypton/blob/master/LICENSE)
```
Remove original BSD Compoenent license part from new file.

Start date for new BSD should be the year created (in this case 2024)

giduac replied:
> mpoenent license part from new file.
> 
> Start date for new BSD should be the year created (in this c

A lot of source files have this Original clause....

Smurf-IV replied:
Original Files - Of course - That is the rule(s) for BSD
New files - have the new start dates etc.

giduac replied:
Corrections posted

giduac commented:
See [source.zip](https://github.com/user-attachments/files/15587406/source.zip)

Added another commit [commit ](https://github.com/Krypton-Suite/Standard-Toolkit/pull/1511/commits/9b719fc604b571f9c36851390f816ed5e8c0ceef) with further changes:

Added: 
- CommonHelperThemeSelectors.InitFromManagerPalette(). Common constructor code.
- CommonHelperThemeSelectors.DefaultPaletteSetter(). Common setter code for DefaultPalette property.

Changed:
- Reset..() and ShouldSerialize...() methods removed from interface and made private.
- The rest of the theme switching logic moved to CommonHelperThemeSelectors
- The control's property DefaultPalette is now updated on a switch via the dropdown list.

![compile-results](https://github.com/Krypton-Suite/Standard-Toolkit/assets/96108132/e43a0124-c7ea-49be-a245-bf5fea8ef2c7)

giduac commented:
@Smurf-IV,

A question: is the themelistbox going to be remved since it's removed from the toolbox?

```cs
namespace Krypton.Toolkit
{
    /// <summary>
    /// Allows the user to change themes using a <see cref="KryptonListBox"/>.
    /// </summary>
    [ToolboxItem(false)]
    [Designer(typeof(KryptonThemeBrowserDesigner))]
    public class KryptonThemeListBox : KryptonListBox
```

Smurf-IV commented:
> @Smurf-IV,
> 
> A question: is the themelistbox going to be removed since it's been removed from the toolbox?
> 
> ```cs
> namespace Krypton.Toolkit
> {
>     /// <summary>
>     /// Allows the user to change themes using a <see cref="KryptonListBox"/>.
>     /// </summary>
>     [ToolboxItem(false)]
>     [Designer(typeof(KryptonThemeBrowserDesigner))]
>     public class KryptonThemeListBox : KryptonListBox
> ```

These controls where created by @Wagnerp , so I'll let him answer, but my vote would be yes.

Smurf-IV commented on Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonThemeComboBox.cs:
```diff
@@ -1,9 +1,12 @@
 ﻿#region BSD License
 /*
- * 
+ *
```
@Wagnerp When did you create this control. It's not part of the Original Toolkit ??

Wagnerp replied:
That would have been last year. It can be removed if necessary.

giduac replied:
@Wagnerp 
That's OK, I'll change the license header on the go....

Wagnerp commented:
> @Smurf-IV,
> 
> A question: is the themelistbox going to be removed since it's been removed from the toolbox?
> 
> ```cs
> namespace Krypton.Toolkit
> {
>     /// <summary>
>     /// Allows the user to change themes using a <see cref="KryptonListBox"/>.
>     /// </summary>
>     [ToolboxItem(false)]
>     [Designer(typeof(KryptonThemeBrowserDesigner))]
>     public class KryptonThemeListBox : KryptonListBox
> ```

Hi @giduac & @Smurf-IV 

I remember now... the reason why it's hidden was due to it only being used in the theme browser UI. It can be removed, or opened up for people to use.

giduac commented:
> > @Smurf-IV,
> > A question: is the themelistbox going to be removed since it's been removed from the toolbox?
> > ```cs
> > namespace Krypton.Toolkit
> > {
> >     /// <summary>
> >     /// Allows the user to change themes using a <see cref="KryptonListBox"/>.
> >     /// </summary>
> >     [ToolboxItem(false)]
> >     [Designer(typeof(KryptonThemeBrowserDesigner))]
> >     public class KryptonThemeListBox : KryptonListBox
> > ```
> 
> Hi @giduac & @Smurf-IV
> 
> I remember now... the reason why it's hidden was due to it only being used in the theme browser UI. It can be removed, or opened up for people to use.

Well I actually like it.
That makes it a draw for now.

So it's up to you @Wagnerp. Thumbs up or down ? :)

giduac commented:
[KThemeCBB license header updated](https://github.com/Krypton-Suite/Standard-Toolkit/pull/1511/commits/199d42329cde3d188ef43cdf11972fe07105f994).

Wagnerp commented:
> > > @Smurf-IV,
> > > A question: is the themelistbox going to be removed since it's been removed from the toolbox?
> > > ```cs
> > > namespace Krypton.Toolkit
> > > {
> > >     /// <summary>
> > >     /// Allows the user to change themes using a <see cref="KryptonListBox"/>.
> > >     /// </summary>
> > >     [ToolboxItem(false)]
> > >     [Designer(typeof(KryptonThemeBrowserDesigner))]
> > >     public class KryptonThemeListBox : KryptonListBox
> > > ```
> > 
> > Hi @giduac & @Smurf-IV
> > 
> > I remember now... the reason why it's hidden was due to it only being used in the theme browser UI. It can be removed, or opened up for people to use.
> 
> Well I actually like it.
> That makes it a draw for now.
> 
> So it's up to you @Wagnerp. Thumbs up or down ? :)

I would say keep it
